test/build/preload-css/__snapshots__

```diff
@@ -0,0 +1,135 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
+
+exports[`snowpack build preload-css: allFiles 1`] = `
+Array [
+  "__snowpack__/env.js",
+  "__snowpack__/optimize-manifest.json",
+  "_dist_/dynamic.css",
+  "_dist_/global-2.css",
+  "_dist_/global-2.css.proxy.js",
+  "_dist_/global.css",
+  "_dist_/global.css.proxy.js",
+  "_dist_/ignored.css",
+  "_dist_/scoped-scss.module.css",
+  "_dist_/scoped-scss.module.css.proxy.js",
+  "_dist_/scoped.module.css",
+  "_dist_/scoped.module.css.proxy.js",
+  "_dist_/vanilla.js",
+  "imported-styles.css",
+  "index.html",
+  "reset.css",
+  "web_modules/import-map.json",
+  "web_modules/water.css/out/water.min.css",
+  "web_modules/water.css/out/water.min.css.proxy.js",
+]
+`;
+
+exports[`snowpack build preload-css: build/__snowpack__/env.js 1`] = `"export default {\\"MODE\\":\\"production\\",\\"NODE_ENV\\":\\"production\\",\\"SSR\\":false};"`;
+
+exports[`snowpack build preload-css: build/__snowpack__/optimize-manifest.json 1`] = `
+"{
+  \\"imports\\": {
+    \\"/index.html\\": {
```

Comment:
…this. This `optimize-manifest.json` first creates an import map, from all HTML files -> CSS and JS. So we can trace any entry file into the things it imports. We needed to build this to see if any CSS proxy files were generated or not.

If we didn’t generate this, then we’d be fighting race conditions in the parallel builder. We would be finding these proxy files, and trying to assemble them, then going back and re-finding all the HTML to inject it in after we’d already finished building everything.

Reply:
+1, this is something that I really want to build on in follow up PRs. We do a lot of work to scan and analyze your application, we should be able to share that result with other optimize steps.